--- cc/private/toolchain/unix_cc_toolchain_config.bzl
+++ cc/private/toolchain/unix_cc_toolchain_config.bzl
@@ -863,6 +863,10 @@
                                         "-rpath",
                                         "-Xlinker",
                                         "$ORIGIN/%{runtime_library_search_directories}",
+                                        "-Xlinker",
+                                        "-z",
+                                        "-Xlinker",
+                                        "origin",
                                     ],
                                 ),
                             ],
